dotnet/ClientLib/ISemanticMemoryClient.cs

```diff
@@ -66,6 +66,21 @@ public interface ISemanticMemoryClient
         string? index = null,
         CancellationToken cancellationToken = default);
 
+
+    /// <summary>
+    /// Check if a document is supported for import.
+    /// </summary>
+    /// <param name="fileName">File name used to detect the file type</param>
+    /// <param name="cancellationToken">Async task cancellation token</param>
+    /// <returns>True if the document is supported for import.</returns>
+    /// <remarks>
+    /// Useful for early validation UX to short-cut queuing an invalid document for import
+    /// and provide more actionable and timely message.
+    /// </remarks>
+    public Task<bool> IsDocumentSupportedAsync(
```

Comment:
A document is a "collection of files" so the name is a bit off. In any case I think these needs to be reconsidered, see other comment in `Program.cs`. There are many reasons that would lead to a false positive or false negative.
Things we could consider (separate PR): when asking if a mime type is supported, pass the list of handlers that will be used, and ask the handlers themselves. Consider that even if supported, file might be invalid. Consider that file name extension might be missing, this is common on Linux/MacOS. Consider the later plan to use file header inspection.

Reply:
I changed this to `GetDocumentTypesAsync`.  While this may still suffer from being disposable (based on moving to content detection), it does fix the "collection" of files aspect.  

If you really still don't like, maybe we could take it off the interface and CC could (as a hack) down cast the client.

Ultimately, I'd like like to avoid maintaining a separate list of supported files in CC.